File: frontend/src/types/media.ts
export interface MediaFormat {
  format_id: string
  type: string
  quality: string
  ext: string | null
  width: number | null
  height: number | null
  fps: number | null
  filesize: number | null
  has_audio: boolean
}

export interface MediaInfo {
  title: string | null
  platform: string | null
  duration: number | null
  thumbnail: string | null
  uploader: string | null
  formats: MediaFormat[]
}

File: backend/app/schemas/media.py
from pydantic import BaseModel, HttpUrl


class MediaInfoRequest(BaseModel):
    url: HttpUrl


class MediaFormatResponse(BaseModel):
    format_id: str
    type: str
    quality: str
    ext: str | None
    width: int | None
    height: int | None
    fps: float | None
    filesize: int | None
    has_audio: bool


class MediaInfoResponse(BaseModel):
    title: str | None
    platform: str | None
    duration: float | None
    thumbnail: str | None
    uploader: str | None
    formats: list[MediaFormatResponse]